### ISSUE TITLE
bit accuracy of wavelets/wfilt_{db,sym}.m looks not so good

### ISSUE THREAD

**Issue by hiyuh** [ltfat/ltfat#99]
i'm now digging `wavelets/wfilt_db.m`.
please see following attached image and linked gist for detail.
after some investigating, i realized this b/c large `N` produces very small `a`.
any thoughts?
![cmp_db](https://user-images.githubusercontent.com/96903/52200364-eea7be00-28ab-11e9-8fc0-9506da889d87.png)
[cmp_db.m](https://gist.github.com/hiyuh/d7a358551bcef59f66ae8e9641aa01a7)

**Comment by susnak** [comment]
Hi,

thanks for looking into this and yes, for large orders, the filters are 
not precise.

The root of the problem is the precision of function roots, which is bad 
for large order polynomials.

The only solution I see is to include precomputed filters as in 
pywavelet. It requires an extended precision, which is not supported in 
Matlab.

Zdenek

**Comment by hiyuh** [comment]
i tried to regenerate `db1` to `db38` coefficients by `boost::multiprecision + eigen3` and it looks good.
please see following attached image and linked gist for detail.
would you make `wfilt_db` to be hybrid of correctly rounded LUT for `db1` to `db38` and current runtime calculus for others?
[coe_db.cxx](https://gist.github.com/hiyuh/b7509b32ee189ffd3c67f75a24c36d00)
[cmp_db.m](https://gist.github.com/hiyuh/d7a358551bcef59f66ae8e9641aa01a7)
![cmp_db](https://user-images.githubusercontent.com/96903/52405392-84db1e80-2b0e-11e9-8f17-7bde59a9a091.png)

**Comment by hiyuh** [comment]
investigating `sym1` to `sym20` too.
[cmp_sym.m](https://gist.github.com/hiyuh/875d06e046643c6dd7681dee61d45d84)
![cmp_sym](https://user-images.githubusercontent.com/96903/52467325-7a7d5b00-2bc8-11e9-8c10-f72706a8326b.png)

**Comment by nholighaus** [comment]
Hey, great that you're looking into this. I'm afraid the wavelet toolbox is the part of LTFAT that I'm least familiar with, so I'll leave it to Zenek to discuss this with you. Nonetheless wanted to say thank you for sharing your findings.Cheers,Nicki-------- Originalnachricht --------Betreff: Re: [ltfat/ltfat] bit accuracy of wavelets/wfilt_{db,sym}.m looks not so good (#99)Von: KIMURA Masaru An: ltfat/ltfat Cc: Subscribed investigating sym1 to sym20 too.
cmp_sym.m


—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/ltfat/ltfat","title":"ltfat/ltfat","subtitle":"GitHub repository","main_image_url":"https://github.githubassets.com/images/email/message_cards/header.png","avatar_image_url":"https://github.githubassets.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/ltfat/ltfat"}},"updates":{"snippets":[{"icon":"PERSON","message":"@hiyuh in #99: investigating `sym1` to `sym20` too.\r\n[cmp_sym.m](https://gist.github.com/hiyuh/875d06e046643c6dd7681dee61d45d84)\r\n![cmp_sym](https://user-images.githubusercontent.com/96903/52467325-7a7d5b00-2bc8-11e9-8c10-f72706a8326b.png)\r\n"}],"action":{"name":"View Issue","url":"https://github.com/ltfat/ltfat/issues/99#issuecomment-461729765"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/ltfat/ltfat/issues/99#issuecomment-461729765",
"url": "https://github.com/ltfat/ltfat/issues/99#issuecomment-461729765",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by susnak** [comment]
@hiyuh yes, that is exactly the same problem with the precision of roots. It makes a lot of sense to just replace everything with pre-computed filters.

@Nicki your email client appends a lot of garbage text in the reply. Could you please check if there is something you can do about that? Thanks

**Comment by hiyuh** [comment]
@nholighaus ack and yw.

@susnak ok. but could you mind to leave runtime computing code for reference as i mentioned as "hybrid"? b/c the code is rare to me.

**Comment by hiyuh** [comment]
long time no see.

investigating `sym*` is slowly reactivated on my `sym` branch.
after reorganize `wfilt_sym.m`, estimated bit accuracy is now varied a bit from `master`,
but doesn't looks so bad.
(maybe, due to reciprocal/conjugate identities and `conv` ordering?)
i'll post some messages after rewriting in `boost::multiprecision + eigen3` later.
anyway, here is current estimated bit accuracy comparison on my `sym` branch.
![cmp_sym](https://user-images.githubusercontent.com/96903/60380246-41f9f300-9a7d-11e9-88d0-12f17034aff5.png)
